scanner.py

```diff
@@ -442,7 +447,23 @@ def start_touch(self, gcmd, touch_settings, verbose):
                     if not has_shown_retry_info:
                         gcmd.respond_info(f"Retry Attempt {int(retries)}")
                         has_shown_retry_info = True  # Set flag to True after showing the message
-
+                
+                if randomize > 0 and new_retry == 1:
+                        # Generate random offsets within ±5 units for both x and y
+                        x_offset = random.uniform(-randomize, randomize)
+                        y_offset = random.uniform(-randomize, randomize)
+                        
+                        cur_pos = self.toolhead.get_position()[:]
+                        
+                        # Apply the random offset to your touch location (assume x, y are current coordinates)
+                        initial_position[0] = initial_position[0] + x_offset
+                        initial_position[1] = initial_position[1] + y_offset
```

Comment:
This looks to me like we're mutating our touch position every time without keeping track of the origin.
What happens if we choose an x offset of 10 two times in a row? Did we move to origin + 20 then?
If that's the case, then I think we should keep track of the origin and apply the offset to that so we ensure movements stay within an area.

Reply:
Ahhh I think I see what you mean.

Reply:
done